lib/addoninfo.cpp

```diff
@@ -0,0 +1,132 @@
+/*
+ * Cppcheck - A tool for static C/C++ code analysis
+ * Copyright (C) 2007-2023 Cppcheck team.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "addoninfo.h"
+
+#include "path.h"
+#include "utils.h"
+
+#include <fstream>
+#include <sstream>
+#include <string>
+
+#include "json.h"
+
+static std::string getFullPath(const std::string &fileName, const std::string &exename) {
+    if (Path::isFile(fileName))
+        return fileName;
+
+    const std::string exepath = Path::getPathFromFilename(exename);
+    if (Path::isFile(exepath + fileName))
+        return exepath + fileName;
+    if (Path::isFile(exepath + "addons/" + fileName))
+        return exepath + "addons/" + fileName;
+
+#ifdef FILESDIR
+    if (Path::isFile(FILESDIR + ("/" + fileName)))
+        return FILESDIR + ("/" + fileName);
+    if (Path::isFile(FILESDIR + ("/addons/" + fileName)))
+        return FILESDIR + ("/addons/" + fileName);
+#endif
+    return "";
+}
+
+static std::string parseAddonInfo(AddonInfo& addoninfo, const picojson::value &json, const std::string &fileName, const std::string &exename) {
+    const std::string& json_error = picojson::get_last_error();
+    if (!json_error.empty()) {
+        return "Loading " + fileName + " failed. " + json_error;
+    }
+    if (!json.is<picojson::object>())
+        return "Loading " + fileName + " failed. Bad json.";
+    picojson::object obj = json.get<picojson::object>();
+    if (obj.count("args")) {
+        if (!obj["args"].is<picojson::array>())
+            return "Loading " + fileName + " failed. args must be array.";
+        for (const picojson::value &v : obj["args"].get<picojson::array>())
+            addoninfo.args += " " + v.get<std::string>();
+    }
+
+    if (obj.count("ctu")) {
+        // ctu is specified in the config file
+        if (!obj["ctu"].is<bool>())
+            return "Loading " + fileName + " failed. ctu must be boolean.";
+        addoninfo.ctu = obj["ctu"].get<bool>();
+    } else {
+        addoninfo.ctu = false;
+    }
+
+    if (obj.count("python")) {
+        // Python was defined in the config file
+        if (obj["python"].is<picojson::array>()) {
```

Comment:
I have the feeling this should say `if (!obj["python"].is<std::string>())` instead .. I have no idea why the old code checked for the array instead :-(

Reply:
Good catch. Will have a look and add some tests.

Reply:
The code made sure it is *not* an array. So that is fine albeit a rather specific handling. Will still adjust though.

Reply:
This requires a lot of new tests to be written. I will do this in a follow-up PR so this can be merged.